### PR TITLE
Add PythonDocs PreCommit job

### PR DESCRIPTION
**Pull request by Amar3tto**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job_PreCommit_PythonDocs.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+jobs:
+  beam_PreCommit_PythonDocs:
```
Comment (andreydevyatkin) [inline .github/workflows/job_PreCommit_PythonDocs.yml:48]: Can you add the name attribute:
`name: beam_PreCommit_PythonDocs`

File: .github/workflows/job_PreCommit_PythonDocs.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (andreydevyatkin) [inline .github/workflows/job_PreCommit_PythonDocs.yml:1]: This file should be named as `beam_PreCommit_PythonDocs.yml`. I noticed there are other python workflow files that we need to rename as part of a separate PR.

File: .github/workflows/job_PreCommit_PythonDocs.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+on:
+  pull_request_target:
+    branches: [ "master", "release-*" ]
+    paths: ["sdks/python/**"]
+  issue_comment:
+    types: [created]
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths: ["sdks/python/**",".github/workflows/job_PreCommit_PythonDocs.yml"]
```
Comment (andreydevyatkin) [inline .github/workflows/job_PreCommit_PythonDocs.yml:27]: See the comment above

**Review by andreydevyatkin — COMMENTED** [review]
see the inline comments

File: .github/workflows/job_PreCommit_PythonDocs.yml (hunk 1 above)
Comment (Amar3tto) [reply .github/workflows/job_PreCommit_PythonDocs.yml]: Done

File: .github/workflows/job_PreCommit_PythonDocs.yml (hunk 3 above)
Comment (Amar3tto) [reply .github/workflows/job_PreCommit_PythonDocs.yml]: Done

File: .github/workflows/beam_PreCommit_PythonDocs.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+jobs:
+  beam_PreCommit_PythonDocs:
+    name: beam_PreCommit_PythonDocs
+    if: |
+      github.event_name == 'push' || 
+      github.event_name == 'pull_request_target' || 
+      github.event.comment.body == 'Run PythonDocs PreCommit' || 
+      github.event_name == 'schedule'
+    runs-on: [self-hosted, ubuntu-20.04]
```
Comment (andreydevyatkin) [inline .github/workflows/beam_PreCommit_PythonDocs.yml:54]: Add the `main` label

Comment (Amar3tto) [reply .github/workflows/beam_PreCommit_PythonDocs.yml]: Done

**Review by Abacn — APPROVED** [review]
Thanks, LGTM

**Pull request merged**